packages/core-accountsjs/core-accountsjs.js

```diff
@@ -0,0 +1,203 @@
+import { AccountsServer, ServerHooks, AccountsJsError } from '@accounts/server';
+import { AccountsPassword } from '@accounts/password';
+import MongoDBInterface from '@accounts/mongo';
+import { MongoInternals } from 'meteor/mongo';
+import { DatabaseManager } from '@accounts/database-manager';
+import { Random } from 'meteor/random';
+import crypto, { randomBytes } from 'crypto';
+import pick from 'lodash.pick';
+import defer from 'lodash.defer';
+
+// import { destroyToken, hashLoginToken } from './util';
+
+const CreateUserErrors = {
+  /**
+   * Will throw if no username or email is provided.
+   */
+  UsernameOrEmailRequired: 'UsernameOrEmailRequired',
+  /**
+   * Username validation via option `validateUsername` failed.
+   */
+  InvalidUsername: 'InvalidUsername',
+  /**
+   * Email validation via option `validateEmail` failed.
+   */
+  InvalidEmail: 'InvalidEmail',
+  /**
+   * Password validation via option `validatePassword` failed.
+   */
+  InvalidPassword: 'InvalidPassword',
+  /**
+   * Email already exist in the database.
+   */
+  EmailAlreadyExists: 'EmailAlreadyExists',
+  /**
+   * Username already exist in the database.
+   */
+  UsernameAlreadyExists: 'UsernameAlreadyExists',
+};
+
+const METEOR_ID_LENGTH = 17;
+
+export const idProvider = () =>
```

Comment:
Why did you replicate the id generation?

I think it's not important that the _ids are created in the same way as Meteor because we don't have to create the id's on the client side, the default for MongoDB is propably a  better choice (ObjectId).

Reply:
Early on during the implementation https://github.com/lorensr/apollo-accounts was used a guidance and I wanted to play it as safe as possible by sticking to Meteor's way of doing things. Though, if it's ok with you we can surely change it.

Reply:
Ok